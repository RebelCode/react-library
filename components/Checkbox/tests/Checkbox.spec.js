import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Checkbox } from "../index";

afterEach(cleanup);

describe("Checkbox", () => {
  it("should take a snapshot", () => {
    const { asFragment } = render(<Checkbox onChange={() => {}} />);

    expect(asFragment(<Checkbox onChange={() => {}} />)).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { queryByLabelText } = render(<Checkbox onChange={() => {}} />);

    expect(queryByLabelText("Checkbox")).toBeTruthy();
  });

  it("renders text correctly", () => {
    const { queryByLabelText } = render(
      <Checkbox onChange={() => {}} />
    );

    expect(queryByLabelText("Checkbox")).toBeTruthy();
  });
});
