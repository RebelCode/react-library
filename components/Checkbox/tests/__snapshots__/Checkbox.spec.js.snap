// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should take a snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Checkbox"
    style="display: flex; flex-flow: row nowrap; justify-content: flex-start; align-items: center;"
  >
    <div
      aria-label="FlexColumn"
      style="display: flex; flex-flow: column nowrap; justify-content: flex-start; align-items: center;"
    >
      <input
        type="checkbox"
        value="1"
      />
    </div>
  </div>
</DocumentFragment>
`;
